src/main/java/org/jabref/gui/fieldeditors/LinkedFilesEditor.java

```diff
@@ -255,14 +255,17 @@ private ContextMenu createContextMenuForFile(LinkedFileViewModel linkedFile) {
         MenuItem renameFile = new MenuItem(Localization.lang("Rename file"));
         renameFile.setOnAction(event -> linkedFile.rename());
         renameFile.setDisable(linkedFile.getFile().isOnlineLink());
+        renameFile.setDisable(linkedFile.performNameConflictCheck());
 
         MenuItem moveFile = new MenuItem(Localization.lang("Move file to file directory"));
         moveFile.setOnAction(event -> linkedFile.moveToDefaultDirectory());
         moveFile.setDisable(linkedFile.getFile().isOnlineLink());
+        moveFile.setDisable(linkedFile.performNameConflictCheck());
```

Comment:
For the `rename file` functionality, the check is ok. However, for `move file` and `rename and move file` the check needs o be adapted: We only want to disable the "move file" option if the file is already in the right place (i.e. the `moveToDefaultDirectory` method would do nothing). In other words, this check should not compare the file name but the directory with the suggested directory.
This was probably my fault for not being clear enough in the issue description.